### PR TITLE
security: `set-output` cmd deprecated. Use `$GITHUB_OUTPUT` env file

### PR DESCRIPTION
**Pull request by davorpa**
## What does this PR do?
Improve repo

### Description

>To avoid untrusted logged data to use `save-state` and `set-output` workflow commands without the intention of the workflow author we have introduced a new set of environment files to manage state and output.

Starting 1st June 2023 workflows using `save-state` or `set-output` commands via stdout will fail with an error.

More info:
- https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
- https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter

### Why is this valuable (or not)?

image

It patches the deprecation

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by davorpa** [comment]
Doing some tests

**Comment by davorpa** [comment]
> Doing some tests to know how to fix awesomebot GitHub summary report
> 
> Compare:
> 
> * OK: [EbookFoundation/free-programming-books/actions/runs/4228037900](https://github.com/EbookFoundation/free-programming-books/actions/runs/4228037900)
> * Bugged: [EbookFoundation/free-programming-books/actions/runs/4242565403](https://github.com/EbookFoundation/free-programming-books/actions/runs/4242565403)

It works again: https://github.com/EbookFoundation/free-programming-books/actions/runs/4243200736

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -81,7 +81,7 @@ jobs:
           stale-pr-label: " "
 
       - name: Print outputs for issues
-        run: echo ${{ join(steps.stale-issues.outputs.*, ',') }}
```
Comment (Thenlie) [inline .github/workflows/stale.yml:84]: Is this change related to the `set-output` deprecation?

Comment (Thenlie) [reply .github/workflows/stale.yml:84]: I see now this is from https://github.com/actions/stale/pull/859

**Review by Thenlie — APPROVED** [review]
This all looks good to me. ✅ 
- No instances of `set-output` or `save-state` were missed in the workflow files. Verified with spotlight search of the repo. 
- Workflow checks are passing and output is properly formatted.
My only nit would be to squash these test commits into one.

File: .github/workflows/stale.yml (hunk 1 above)
Comment (davorpa) [reply .github/workflows/stale.yml:84]: Not is related, but since it is a superfluous change, I have taken advantage of this PR to do it.

It should be do it past in #9242 bump
